cpp/src/groupby/hash/groupby.cu

```diff
@@ -512,18 +512,29 @@ rmm::device_uvector<size_type> extract_populated_keys(map_type const& map,
 {
   rmm::device_uvector<size_type> populated_keys(num_keys, stream);
 
-  auto get_key    = [] __device__(auto const& element) { return element.first; };  // first = key
-  auto get_key_it = thrust::make_transform_iterator(map.data(), get_key);
-  auto key_used   = [unused = map.get_unused_key()] __device__(auto key) { return key != unused; };
-
-  auto end_it = thrust::copy_if(rmm::exec_policy(stream),
-                                get_key_it,
-                                get_key_it + map.capacity(),
-                                populated_keys.begin(),
-                                key_used);
-
-  populated_keys.resize(std::distance(populated_keys.begin(), end_it), stream);
+  auto get_key  = [] __device__(auto const& element) { return element.first; };  // first = key
+  auto key_itr  = thrust::make_transform_iterator(map.data(), get_key);
+  auto key_used = [unused = map.get_unused_key()] __device__(auto key) { return key != unused; };
+
+  // thrust::copy_if has a bug where it cannot iterate over int-max values
+  // so if map.capacity() > int-max we'll call thrust::copy_if in chunks instead
+  auto const map_size =
```

Comment:
nit, could this be changed to something like "copy_size"? "map_size" seems to indicate that it is the size of the actual map.

Reply:
It is the size of the map. We are iterating over the entire map to copy out the keys.

Reply:
I agree that this is the map size in the regular scenario, where `map.capacity()` is less than or equal to `std::numeric_limits<size_type>::max()`, but in the case where it is larger, it is the the chunk of the map we are going to copy, not the whole size.

Reply:
I agree that the name should coney this is the max length we can pass to thrust::copy_if